Objects/Converters/ConverterAutocadCivil/ConverterAutocadCivilShared/ConverterAutocadCivil.Civil.cs

```diff
@@ -267,52 +264,52 @@ public ApplicationObject AlignmentToNative(Alignment alignment)
         GetFromObjectIdCollection(civilAlignment["label"] as string, labelStyles, true) : civilDoc.Styles.LabelSetStyles.AlignmentLabelSetStyles.First();
 #endregion
 
-      try
+      // create the alignment
+      var id = ObjectId.Null;
+      switch (type)
       {
-        // add new alignment to doc
-        // ⚠ this will throw if name is not unique!!
-        var id = ObjectId.Null;
-        switch (type)
-        {
-          case AlignmentType.Offset:
-            // create only if parent exists in doc
-            if (parent == ObjectId.Null)
-            {
-              goto default;
-            }
+        case AlignmentType.Offset:
+          // create only if parent exists in doc
+          if (parent == ObjectId.Null || civilAlignment.offset == 0)
+          {
+            goto default;
+          }
 
+          try
+          {
+            id = CivilDB.Alignment.CreateOffsetAlignment(name, parent, civilAlignment.offset, style);
+          }
+          catch (ArgumentException) // throws when name is invalid or offset is too large
+          {
+            // try to recreate with a unique name
             try
-            {
-              id = CivilDB.Alignment.CreateOffsetAlignment(name, parent, civilAlignment.offset, style);
-            }
-            catch
             {
               id = CivilDB.Alignment.CreateOffsetAlignment(CivilDB.Alignment.GetNextUniqueName(name), parent, civilAlignment.offset, style);
             }
-            break;
-          default:
-            try
-            {
-              id = CivilDB.Alignment.Create(civilDoc, name, site, layer, style, label);
-            }
-            catch
+            catch (ArgumentException)
             {
-              id = CivilDB.Alignment.Create(civilDoc, CivilDB.Alignment.GetNextUniqueName(name), site, layer, style, label);
+              goto default;
```

Comment:
Really not keen on this. I'd suggest adding a CreateDefaultValue() private method and calling it from both places.